src/sdk/main/include/TokenCreateTransaction.h

```diff
@@ -167,9 +173,20 @@ class TokenCreateTransaction : public Transaction<TokenCreateTransaction>
    */
   TokenCreateTransaction& setAutoRenewAccountId(const AccountId& accountId);
 
+  /**
+   * Assignment logic of the autoRenewAccount for a transaction.
+   *
+   * @param body The body of the transaction that requires custom AutoRenewAccount assignment logic.
+   * @param accountId The auto renew account to be assigned.
+   */
+  void assignAutoRenewAccount(proto::TransactionBody& body, const AccountId& accountId) const override;
```

Comment:
Should this be declared as `protected` instead of `public`? Since it's only used internally by `Transaction`?